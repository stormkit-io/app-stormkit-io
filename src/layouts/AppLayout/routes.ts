import Async from "~/components/Async";

export const routes = [
  {
    path: "/environments",
    element: Async(() => import("~/pages/apps/[id]/environments")),
  },
  {
    path: "/environments/default",
    element: Async(() => import("~/pages/apps/[id]/environments/Default")),
  },
  {
    path: "/environments/:envId/*",
    element: Async(() => import("~/pages/apps/[id]/environments/[env-id]")),
  },
  {
    path: `/settings`,
    element: Async(() => import("~/pages/apps/[id]/settings")),
  },
  {
    path: `/deployments`,
    element: Async(() => import("~/pages/apps/[id]/deployments")),
  },
  {
    path: "/deployments/:deploymentId",
    element: Async(
      () => import("~/pages/apps/[id]/deployments/[deployment-id]")
    ),
  },
  {
    path: `/usage`,
    element: Async(() => import("~/pages/apps/[id]/usage")),
  },
];
