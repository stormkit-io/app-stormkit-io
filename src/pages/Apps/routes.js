import Async from "~/components/Async";

export default [
  {
    path: "/apps/:id",
    exact: true,
    component: Async(() => import("~/pages/Apps/:id")),
  },
  {
    path: "/apps/:id/environments",
    component: Async(() =>
      import("~/pages/Apps/:id/Environments/Environments.context")
    ),
  },
  {
    path: "/apps/:id/deployments",
    exact: true,
    component: Async(() => import("~/pages/Apps/:id/Deployments")),
  },
  {
    path: "/apps/:id/deployments/:deploymentId",
    exact: true,
    component: Async(() =>
      import("~/pages/Apps/:id/Deployments/:deploymentId")
    ),
  },
  {
    path: "/apps/:id/team",
    exact: true,
    component: Async(() => import("~/pages/Apps/:id/Team")),
  },
  {
    path: "/apps/:id/settings",
    exact: true,
    component: Async(() => import("~/pages/Apps/:id/Settings")),
  },
];
