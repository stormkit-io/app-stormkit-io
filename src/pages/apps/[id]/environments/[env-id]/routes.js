import React from "react";
import Async from "~/components/Async";
import { Redirect } from "react-router-dom";

export default [
  {
    path: "/apps/:id/environments/:envId",
    exact: true,
    component: ({ match }) => <Redirect to={`${match.url}/logs`} />,
  },
  {
    path: "/apps/:id/environments/:envId/logs",
    component: Async(() =>
      import("~/pages/apps/[id]/environments/[env-id]/logs")
    ),
  },
  {
    path: "/apps/:id/environments/:envId/remote-config",
    component: Async(() =>
      import("~/pages/apps/[id]/environments/[env-id]/remote-config")
    ),
  },
  {
    path: "/apps/:id/environments/:envId/snippets",
    component: Async(() =>
      import("~/pages/apps/[id]/environments/[env-id]/snippets")
    ),
  },
  {
    path: "/apps/:id/environments/:envId/domain",
    component: Async(() =>
      import("~/pages/apps/[id]/environments/[env-id]/domain")
    ),
  },
];
