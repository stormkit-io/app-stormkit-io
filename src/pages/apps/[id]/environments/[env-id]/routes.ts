import type { RouteProps } from "react-router";
import Async from "~/components/Async";

const routes: Array<RouteProps> = [
  {
    path: "/",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/config")
    ),
  },
  {
    path: "/logs",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/logs")
    ),
  },
  {
    path: "/feature-flags",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/feature-flags")
    ),
  },
  {
    path: "/snippets",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/snippets")
    ),
  },
  {
    path: "/deployments",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/deployments")
    ),
  },
  {
    path: "/deployments/:deploymentId",
    element: Async(
      () =>
        import(
          "~/pages/apps/[id]/environments/[env-id]/deployments/[deployment-id]"
        )
    ),
  },
  {
    path: "/deployments/:deploymentId/runtime-logs",
    element: Async(
      () =>
        import(
          "~/pages/apps/[id]/environments/[env-id]/deployments/runtime-logs"
        )
    ),
  },
];

export default routes;
