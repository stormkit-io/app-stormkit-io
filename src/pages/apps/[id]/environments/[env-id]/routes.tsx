import type { RouteProps } from "react-router";
import React from "react";
import { Navigate } from "react-router-dom";
import Async from "~/components/Async";

const routes: Array<RouteProps> = [
  {
    path: "/",
    element: <Navigate to={`logs`} />,
  },
  {
    path: "/logs",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/logs")
    ),
  },
  {
    path: "/feature-flags",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/feature-flags")
    ),
  },
  {
    path: "/snippets",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/snippets")
    ),
  },
  {
    path: "/domain",
    element: Async(
      () => import("~/pages/apps/[id]/environments/[env-id]/domain")
    ),
  },
];

export default routes;
