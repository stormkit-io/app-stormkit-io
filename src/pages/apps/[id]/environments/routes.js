import Async from "~/components/Async";

export default [
  {
    path: "/apps/:id/environments",
    exact: true,
    component: Async(() => import("~/pages/apps/[id]/environments"))
  },
  {
    path: "/apps/:id/environments/:envId",
    component: Async(() =>
      import("~/pages/apps/[id]/environments/[env-id]/Environment.context")
    )
  }
];
