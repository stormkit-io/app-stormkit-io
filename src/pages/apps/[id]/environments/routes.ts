import Async from "~/components/Async";
import { RouteProps } from "react-router-dom";

const routes: Array<RouteProps> = [
  {
    path: "/apps/:id/environments",
    exact: true,
    component: Async(() => import("~/pages/apps/[id]/environments")),
  },
  {
    path: "/apps/:id/environments/:envId",
    component: Async(
      () =>
        import("~/pages/apps/[id]/environments/[env-id]/Environment.context")
    ),
  },
];

export default routes;
