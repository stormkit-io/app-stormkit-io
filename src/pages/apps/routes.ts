import Async from "~/components/Async";
import { RouteProps } from "react-router-dom";

const routes: Array<RouteProps> = [
  {
    path: "/apps/:id",
    exact: true,
    component: Async(() => import("~/pages/apps/[id]")),
  },
  {
    path: "/apps/:id/environments/:envId",
    component: Async(() => import("~/pages/apps/[id]/environments/[env-id]")),
  },
  {
    path: "/apps/:id/environments",
    component: Async(
      () => import("~/pages/apps/[id]/environments/Environments")
    ),
  },
  {
    path: "/apps/:id/deployments",
    exact: true,
    component: Async(() => import("~/pages/apps/[id]/deployments")),
  },
  {
    path: "/apps/:id/deployments/:deploymentId",
    exact: true,
    component: Async(
      () => import("~/pages/apps/[id]/deployments/[deployment-id]")
    ),
  },
  {
    path: "/apps/:id/team",
    exact: true,
    component: Async(() => import("~/pages/apps/[id]/team")),
  },
  {
    path: "/apps/:id/settings",
    exact: true,
    component: Async(() => import("~/pages/apps/[id]/settings")),
  },
  {
    path: "/apps/:id/usage",
    exact: true,
    component: Async(() => import("~/pages/apps/[id]/usage")),
  },
];

export default routes;
