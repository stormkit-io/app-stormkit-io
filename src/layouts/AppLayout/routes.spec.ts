import { routes } from "./routes";

describe("~/layouts/AppLayout/routes.ts", () => {
  test("should match routes", () => {
    expect(routes).toMatchSnapshot();
  });
});
