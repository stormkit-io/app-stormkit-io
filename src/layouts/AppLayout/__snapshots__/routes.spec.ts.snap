// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`~/layouts/AppLayout/routes.ts should match routes 1`] = `
[
  {
    "element": <React.Suspense
      fallback={
        <Spinner
          pageCenter={true}
        />
      }
    >
      <UNDEFINED />
    </React.Suspense>,
    "path": "/environments",
  },
  {
    "element": <React.Suspense
      fallback={
        <Spinner
          pageCenter={true}
        />
      }
    >
      <UNDEFINED />
    </React.Suspense>,
    "path": "/environments/default",
  },
  {
    "element": <React.Suspense
      fallback={
        <Spinner
          pageCenter={true}
        />
      }
    >
      <UNDEFINED />
    </React.Suspense>,
    "path": "/environments/:envId/*",
  },
  {
    "element": <React.Suspense
      fallback={
        <Spinner
          pageCenter={true}
        />
      }
    >
      <UNDEFINED />
    </React.Suspense>,
    "path": "/settings",
  },
  {
    "element": <React.Suspense
      fallback={
        <Spinner
          pageCenter={true}
        />
      }
    >
      <UNDEFINED />
    </React.Suspense>,
    "path": "/deployments",
  },
  {
    "element": <React.Suspense
      fallback={
        <Spinner
          pageCenter={true}
        />
      }
    >
      <UNDEFINED />
    </React.Suspense>,
    "path": "/deployments/:deploymentId",
  },
  {
    "element": <React.Suspense
      fallback={
        <Spinner
          pageCenter={true}
        />
      }
    >
      <UNDEFINED />
    </React.Suspense>,
    "path": "/usage",
  },
]
`;
